packages/flutter_tools/lib/src/commands/build_apk.dart

```diff
@@ -573,3 +581,15 @@ Future<int> buildApk(
 
   return result;
 }
+
+String _getLastBuildType(String buildDirectoryPath) {
+  File lastBuildFile = new File(path.join(buildDirectoryPath, '.build'));
```

Comment:
I'd suggest making this file visible and using a structured data format (JSON?) in case we need to extend the contents